graphql/resolvers/user/mutations.js

```diff
@@ -0,0 +1,52 @@
+import { User } from "../../../db/index.js";
+import { hash, verify } from "../../../auth/auth.js";
+import jwt from "jsonwebtoken";
+
+export default {
+    login: async (_, { username, password }) => {
+        const result = await User.findOne({ username })
+            .select("+password")
+            .exec();
+        if (result == null) {
+            throw {
+                message: "Username or password incorrect.",
+            };
+        }
+        if (await verify(password, result.password)) {
+            return createUserToken(result, username);
+        } else {
+            throw {
+                message: "Username or password incorrect.",
+            };
+        }
```

Comment:
Return inside if makes else block redundant
```suggestion
        }
        throw {
            message: "Username or password incorrect.",
        };
```